swift/ql/src/queries/Security/CWE-089/SqlInjection.qhelp

```diff
@@ -12,13 +12,13 @@ If a database query (such as a SQL query) is built from user-provided data witho
 <recommendation>
 
 <p>
-Most database connector libraries offer a way to safely embed untrusted data into a query using query parameters or prepared statements. You should use these features to build queries, rather than string concatenation or similar methods without sufficient sanitization.
+Most database connector libraries offer a way to safely embed untrusted data into a query using query parameters or prepared statements. You should use these features to build queries, rather than string concatenation or similar methods. It's also possible to escape (sanitize) user-controlled strings so that they can be included directly in an SQL command, but this approach is only safe if the chosen escaping function is robust.
```

Comment:
```suggestion
Most database connector libraries offer a way to safely embed untrusted data into a query using query parameters or prepared statements. You should use these features to build queries, rather than string concatenation or similar methods. You can also escape (sanitize) user-controlled strings so that they can be included directly in an SQL command, but this approach is only safe if the chosen escaping function is robust.
```

I'm not a subject matter expert so I don't know how valid this is, but to a layman (if laymans would ever read this), some description of what constitutes "robustness" in the context might be useful?

Otherwise I'd just make this small suggestion to make it second-person/active, but it's a fairly weakly held suggestion so up to you

Reply:
The phrasing is kind of designed to acknowledge this approach exists, but not to really encourage it.  When people roll their own sanitization routines they often get tripped up by edge cases, resulting in the code that is still vulnerable.  I think therefore if we recommend something more precise it should be to use an "official" function provided by the SQL library.  I'll see if I can come up with some phrasing...

Reply:
It's definitely not urgent feedback at all so don't spend too long on it on my account!

Reply:
I've pushed a better description (open to further improvements).